Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -3327,6 +3328,13 @@ static inline bool isOrthogonal(const RenderBox& renderer, const RenderElement&
 
 std::optional<LayoutUnit> RenderBox::computeIntrinsicLogicalContentHeightUsing(Length logicalHeightLength, std::optional<LayoutUnit> intrinsicContentHeight, LayoutUnit borderAndPadding) const
 {
+    if (auto* renderImage = dynamicDowncast<RenderImage>(this)) {
+        auto computedLogicalWidth = style().logicalWidth();
+        auto intrinsicRatio = renderImage->intrinsicRatio();
+        if ((logicalHeightLength.isMinContent() || logicalHeightLength.isMaxContent()) && computedLogicalWidth.isFixed() && !style().hasAspectRatio())
+            return resolveHeightForRatio(borderAndPaddingLogicalWidth(), borderAndPaddingLogicalHeight(), LayoutUnit(computedLogicalWidth.value()), intrinsicRatio.transposedSize().aspectRatio(), BoxSizing::ContentBox);
+    }
+
```

Comment:
I'd move this logic under `if (logicalHeightLength.isMinContent() || logicalHeightLength.isMaxContent() ||` at line# 3332